state-chain/pallets/cf-validator/src/lib.rs

```diff
@@ -1218,10 +1218,13 @@ impl<T: Config> Pallet<T> {
 	/// sorted by bids highest to lowest.
 	pub fn highest_funded_qualified_backup_node_bids(
 	) -> impl Iterator<Item = Bid<ValidatorIdOf<T>, <T as Chainflip>::Amount>> {
-		let mut backups: Vec<_> = Backups::<T>::get()
+		let backup_map = Backups::<T>::get();
+		let qualified_backups =
+			T::KeygenQualification::take_qualified(backup_map.keys().cloned().collect());
```

Comment:
both uses of `take_qualified` are then followed by a filter of something else, based on the `qualified` list, perhaps we can factor out further here

Reply:
Yeah I think I have an idea.